aws/src/main/java/org/apache/iceberg/aws/s3/VendedCredentialsProvider.java

```diff
@@ -75,10 +76,13 @@ private RESTClient httpClient() {
     if (null == client) {
       synchronized (this) {
         if (null == client) {
+          DefaultAuthSession authSession =
+              DefaultAuthSession.of(
+                  HTTPHeaders.of(OAuth2Util.authHeaders(properties.get(OAuth2Properties.TOKEN))));
```

Comment:
@adutra It's beyond the scope of this PR since this is an existing problem, but somehow we need the AuthManager/AuthSession to be able to refresh OAuth tokens here as well. The token here will expire at which point the credential provider will not be able to get new vended credentials.

Reply:
Yes. I don't mind looking into it once we have the whole API merged 👍